app.js

```diff
@@ -0,0 +1,132 @@
+// //1 object with array property
+// let myObject = {
+//     thisKey: [1,2,3,4]
+// };
+// console.log(myObject.thisKey[0]);
+//
+// //2 object with inner object
+// let obj = {
+//     good: {
+//         bad: "great"
+//     }
+// };
+// console.log(obj.good.bad);
+//
+// // 3
+// let otherObj = {
+//     doSomething: function (){
+//         console.log("this is a function");
+//     }
+// };
+// otherObj.doSomething();
+//
+// //4 array with object
+// let myArray = [
+//     {cool: "stuff"}
+// ];
+// console.log(myArray[0]);
+//
+// //5 array with array
+// let otherArr = [
+//     [1,2,3]
+// ];
+// console.log(otherArr[0][1]);
+//
+// // 6
+// let arr = [1,2, function(){console.log("this is an array function")}];
+// arr[2]();
+//
+// //7
+// let objArr = {
+//     [1,2,3,4]
+// };
```

Comment:
this won't count against you because you only had to do 3 and you did 6 correctly but this array needs a key
line 41 consequently causes an error.  which suggests that you weren't checking your code (in the console) after each step.  check your code after each step!